### PR TITLE
Add a switch for JDK version

### LINKED ISSUE #70: Add a switch for JDK version
Let the user choose between JDK8 and JDK11

### PR DESCRIPTION
**Pull request by tsegismont**
Fixes #70

### CODE REVIEW

**Comment by tsegismont** [comment]
@jponge @danielpetisme can one of you please take a look?

The PR is marked "do not merge yet" because it's based upon #89. You can review the last commits only.

File: build.gradle.kts [hunk 1]
```diff
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+import org.gradle.api.JavaVersion.VERSION_1_8
 import org.gradle.api.tasks.wrapper.Wrapper.DistributionType.ALL
 
 plugins {
```
Comment (jponge) [inline build.gradle.kts]: The vert.x plugin already applies `java`

File: build.gradle.kts [hunk 2]
```diff
@@ -46,8 +48,13 @@ dependencies {
   testImplementation("io.vertx:vertx-junit5:${vertxVersion}")
   testImplementation("com.julienviet:childprocess-vertx-ext:1.3.0")
 
-  //When present gradle test work but intellij not and vice-versa
-  testRuntime("org.junit.jupiter:junit-jupiter-engine:5.4.0")
+  testImplementation("org.junit.jupiter:junit-jupiter-engine:${junitJupiterVersion}")
+  testImplementation("org.junit.jupiter:junit-jupiter-params:${junitJupiterVersion}")
+}
+
+java {
```
Comment (jponge) [inline build.gradle.kts]: Already done by the vertx plugin

File: build.gradle.kts [hunk 3]
```diff
@@ -31,6 +32,7 @@ version = "2.0.2"
 description = "A web application to generate Vert.x projects"
 
 val vertxVersion = "3.6.3"
+val junitJupiterVersion = "5.4.1"
 
 dependencies {
   implementation("io.vertx:vertx-web:${vertxVersion}")
```
Comment (jponge) [inline build.gradle.kts]: If you set `vertxVersion` in the vertx plugin extension configuration, then you can skip the version in dependencies from the Vert.x stack. It uses the Nebula plugin under the hood.

File: build.gradle.kts (hunk 1 above)
Comment (tsegismont) [reply build.gradle.kts]: Does it apply version 1.8?

**Comment by jponge** [comment]
Yes it does

**Comment by tsegismont** [comment]
@jponge updated as per comments. Can you please take another (quick) look?

**Comment by tsegismont** [comment]
> @tsegismont you are now a Gradle guru wink

Great! Can you validate the PR then? :wink:

**Comment by jponge** [comment]
Great!

**Pull request merged**